services/compute/virtualmachinescaleset/virtualmachinescaleset.go

```diff
@@ -328,13 +345,18 @@ func (c *client) getWssdVirtualMachineScaleSetVMProfile(vmp *compute.VirtualMach
 	if err != nil {
 		return nil, errors.Wrapf(err, "Invalid VMSS VMProfile OS")
 	}
+	guestagent, err := c.getWssdVirtualMachineScaleSetGuestAgentConfiguration(vmp)
+	if err != nil {
```

Comment:
if check not needed

Reply:
The called function should be a black box for the caller, so we still should check for the error, even if it might always be nil